workspaces/local-cli/src/commands/status.ts

```diff
@@ -114,20 +110,37 @@ export default class Status extends Command {
     const { diffs } = await diffService.listDiffs();
     const { urls } = await diffService.listUnrecognizedUrls();
 
-    const requests: any = await spectacle.query({
+    const requestQuery = await spectacle.query<any>({
```

Comment:
This is used in both local-cli and the ui, there's probably a better place to put this, so we don't need to duplicate, but I haven't found a great place for this (maybe the `cli-shared` workspace)?

Reply:
I've also tested `api start` in the sidechannel build https://github.com/opticdev/optic/actions/runs/1016146932 and this works (less certain because of loose typings, but that's more to copy paste)

There's probably a better strategy since it's likely we'll want almost the same functionality in the UI as the local cli

Reply:
While _some_ of the logic might be shared, isn't the entire point of graph ql queries is that they can be and should be client specific? So that any can optimise exactly what it queries for, without worrying about affecting others? I'd be happy for code to be duplicated, especially if it's not particularly error prone. Especially for "glue" code, that's what I'd expect, as even if clients are context are similar, they hardly are ever the same, nor will they stay that way.

Reply:
There's more transformation logic - I think as an ideal, there should be more that moves into spectacle, but I think the more complex our usecases become, we'll have specialized logic on top of that (that shouldn't live in spectacle) that need more complex transformations, etc. Lets say we implement a diff flow in the UI or other FEs, we'd probably want a shared layer between the UI and the cli, but spectacle probably shouldn't take all of these.

I think the biggest pain is that spectacle.query + defining our gql query produces an `any` result and we have to manually type the return, which causes the most pain